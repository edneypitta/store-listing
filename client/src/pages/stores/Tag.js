import React from 'react'
import Emoji from 'react-emoji-render'

const emojis = {
  'american': ':flag_us:',
  'fat': ':doughnut:',
  'healthy': ':running:',
  'breakfast': ':egg:',
  'indian': ':flag_in:',
  'coffee': ':coffee:',
  'watches': ':watch:',
  'chocolate': ':chocolate_bar:',
  'food': ':yum:',
  'burgers': ':hamburger:',
  'pizza': ':pizza:',
  'spicy': 'üå∂Ô∏è',
  'italian': ':flag_it:',
  'vegan': ':seedling:'
}

export default function Tag({ tag, tagFilters, onTagClick }) {
  const filtered = tagFilters.includes(tag)

  return (
    <button
      key={tag}
      className={filtered ? 'filtered' : ''}
      title={filtered ? 'Click to remove filter' : 'Click to filter by tag'}
      value={tag}
      onClick={onTagClick}>
      <Emoji text={emojis[tag]} />
      <span className='tag-name'>{tag}</span>
    </button>
  )
}
