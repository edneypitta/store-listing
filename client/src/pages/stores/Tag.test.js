import React from 'react'
import { render } from 'enzyme'
import Tag from './Tag'

describe('Store', () => {
  it('renders without crashing', () => {
    const component = render(
      <Tag tag='fat' tagFilters={[]} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when filtered', () => {
    const component = render(
      <Tag tag='fat' tagFilters={['fat']} />)

    expect(component).toMatchSnapshot()
  })
})