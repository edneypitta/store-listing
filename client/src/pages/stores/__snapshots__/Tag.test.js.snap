// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store renders when filtered 1`] = `
<button
  class="filtered"
  title="Click to remove filter"
  value="fat"
>
  <span
    class="undefined"
  >
    <span
      style="width:1em;height:1em;margin:0 .05em 0 .1em;vertical-align:-0.1em"
    >
      🍩
    </span>
  </span>
  <span
    class="tag-name"
  >
    fat
  </span>
</button>
`;

exports[`Store renders without crashing 1`] = `
<button
  class=""
  title="Click to filter by tag"
  value="fat"
>
  <span
    class="undefined"
  >
    <span
      style="width:1em;height:1em;margin:0 .05em 0 .1em;vertical-align:-0.1em"
    >
      🍩
    </span>
  </span>
  <span
    class="tag-name"
  >
    fat
  </span>
</button>
`;
